ckan/logic/validators.py

```diff
@@ -858,3 +858,20 @@ def email_validator(value, context):
         if not email_pattern.match(value):
             raise Invalid(_('Email {email} is not a valid format').format(email=value))
     return value
+
+def one_of(value,context):
```

Comment:
We can not use `context` to provide the iterable as there is no way to set it in the first place, and it is used internally by CKAN (in fact we don't need the context parameter at all). What we can do is to provide a wrapper that accepts a parameter, in the same way that for instance [`if_empty_same_as`](https://github.com/ckan/ckan/blob/master/ckan/lib/navl/validators.py#L37) does.

```python
def one_of(list_of_values):
    def callable(value):
         if value not in list_of_values:
             raise Invalid(_('Value must be one of {}'.format(list_of_values)))
         return value
```
This will be then use like this in the schemas:

```python
    {
         'some_custom_field': [ignore_missing, one_of(['bear', 'lion', 'tiger']), unicode_safe),
    }

```